collect_app/src/main/java/org/odk/collect/android/preferences/AutoSendPreferenceMigrator.java

```diff
@@ -0,0 +1,83 @@
+package org.odk.collect.android.preferences;
+
+import android.annotation.SuppressLint;
+import android.content.SharedPreferences;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.Map;
+
+import static org.odk.collect.android.preferences.PreferenceKeys.KEY_AUTOSEND;
+import static org.odk.collect.android.preferences.PreferenceKeys.KEY_AUTOSEND_NETWORK;
+import static org.odk.collect.android.preferences.PreferenceKeys.KEY_AUTOSEND_WIFI;
+
+/**
+ * Migrates existing autosend_wifi and autosend_network preference values to autosend
+ */
+public class AutoSendPreferenceMigrator {
+
+    private static boolean autoSendWifi = false;
```

Comment:
I don't think any of these are needed. You pass the values from the methods with the public interfaces to the private helper which is good. And the variables could just be declared locally.

In general, I try to avoid static fields that are not final unless they very explicitly represent shared state between instances (a shared counter that all instances increment for example).